### ISSUE TITLE
programs.loc should probably be factored out into a locations table

### ISSUE THREAD

**Issue by mattjuk** [nineworldsgeekfest/scheduler#36]
The `loc` field for `programs` was really a shortcut to avoid having a better schema for rooms or locations. If we're doing this right, we should really factor that out into a separate object / table.

**Comment by DanielJohnston** [comment]
Agreed. I left it in rather than revisiting as trying to replicate existing functionality seemed more important, but obvs there's a disconnect between the generalisation to multiple conferences and still having hard-coded rooms. Would suggest this is one for after we get the basic interface and model restrictions working on existing stuff.
